File: data_structure/hash_map/442_Find_All_Duplicates_in_an_Array.py
from collections import Counter

class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        
        nums_cnt = Counter(nums)
        res = []
        for key in nums_cnt:
            if nums_cnt[key] == 2:
                res.append(key)
        
        return res

File: golang/442_FindAllDuplicates_in_anArray.go
func findDuplicates(nums []int) (res []int) {
    seen := make(map[int]int, len(nums)) 

    for _, v := range nums {
        _, ok := seen[v]
        if !ok {
            seen[v] = 1
        } else {
            seen[v]++
        }
    }

    for k, v := range seen {
        if v > 1 {
            res = append(res, k)
        }
    }

    return res
}
